### ISSUE TITLE
Allow decoration of default scheduler

### ISSUE THREAD

**Issue by whiskeysierra** [jhalterman/failsafe#204]
The default scheduler is `DelegatingScheduler.INSTANCE` but it's in an `internal` package. I could just decorate it, e.g. to propagate OpenTracing context, but it's risky because it's not officially part of the API. Ideally there would be a public method that gives me the default `Scheduler`, e.g. `Scheduler.getDefaultScheduler()`.

**Comment by jhalterman** [comment]
So are you thinking of doing something like:

```java
Scheduler scheduler = Scheduler.getDefaultScheduler();
scheduler = instrumentWithTracing(scheduler);

Failsafe.with(policies).with(scheduler).get(it);
```

?

**Comment by whiskeysierra** [comment]
Exactly. I'm open to alternatives of course. Maybe I could just decorate
the task directly. Have to try.

**Comment by jhalterman** [comment]
Released in 2.3.1.
